algorithms/polish.sbl

```diff
@@ -0,0 +1,225 @@
+stringescapes {}
+
+stringdef ak '{U+0105}' // ę e + ogonek
+stringdef ek '{U+0119}' // ą a + ogonek
+stringdef l/ '{U+0142}' // ł l + stroke
+stringdef s' '{U+015b}' // ś s + acute (kreska)
+stringdef c' '{U+0107}' // ć c + acute (kreska)
+stringdef o' '{U+00f3}' // ó o + acute (kreska)
+
+
+externals (stem)
+
+
+routines (
+  remove_nouns
+  remove_dimunitive
+  remove_superlative_ends
+  remove_adjective_ends
+  remove_verbs_ends
+  remove_adverbs_ends
+  remove_plural_forms
+  remove_general_ends
+)
+
+
+backwardmode (
+  define remove_nouns as (
+    (
+      $(len > 6)
+      [substring] among (
+        // Borrowed words that match English -ation, e.g. fortyfikacja (fortification).
+        'acja'     // nominative
+        'acji'     // genitive/dative/locative
+        'acj{ak}'  // accusative
+        // Plural locative ending of (rare) words ending with -ę, e.g. zwierzę - zwierzętach.
+        // TODO: It can also be roots ending with -t and -ach ending, e.g. pirat - piratach.
+        'tach'
+        // Verbal nouns, e.g. czytanie, narodzenie.
+        'anie' 'enie'  // nominative
+        'aniu' 'eniu'  // dative/locative
+          (delete)
+        // Borrowed words that match English -tics, e.g. lingwistyka (linguistics).
+        'tyka'
+          (<- 'ty')
+      )
+    ) or (
+      $(len > 5)
+      [substring] among (
+        'ach'  // plural locative
+        'ami'  // plural instrumental
+        'nia'  // genitive of words ending with -nie, e.g. czytanie - czytania
+        'niu'  // dative/locative of words ending with -nie, e.g. czytanie - czytaniu
+        'cia'  // genitive of words ending with -cie, e.g. ukrycie - ukrycia
+        'ciu'  // dative/locative of words ending with -cie, e.g. ukrycie - ukryciu
+        'ce'   // dative/locative of words ending with -ka, e.g. rzeka - rzece
+        'ta'   // nominative/accusative plural of (rare) words ending with -ę, e.g. zwierzę - zwierzęta
+          (delete)
+        // Borrowed words that match English -tion, e.g. intencja (intention).
+        'cja'     // nominative
+        'cji'     // genitive/dative/locative
+        'cj{ak}'  // accusative
+          (<- 'c')
+      )
+    )
+  )
+
+  define remove_dimunitive as (
+    (
+      $(len > 6)
+      [substring] among (
+        'eczek'  // e.g. człowieczek (diminutive of człowiek)
+        'iczek'  // e.g. czajniczek (diminutive of czajnik)
+        'iszek'  // e.g. braciszek (diminutive of brat)
+        'aszek'  // e.g. młodzieniaszek (diminutive of młodzieniec)
+        'uszek'  // e.g. ‎kwiatuszek (diminutive of ‎kwiat)
+          (delete)
+        'enek'   // e.g. ogienek (diminutive of ogień)
+          (<- 'en')
+        'ejek'   // e.g. olejek (diminutive of olej)
```

Comment:
The example fails the `$(len > 6)` requirement above.

Reply:
However `olejek` is stemmed to `olej` by the `ek` rule in the group below, so this rules seems to be unnecessary (same for the `enek` and `erek` rules before/after it).